### PR TITLE
Rider and driver images

### PR DESCRIPTION
**Pull request by helenxuyang**
### Summary 

This pull request is the first step towards implementing profile images for riders and drivers

- [x] display images from backend for riders and driver
- [x] added profile picture selection on the profile page

### Test Plan 
Tested on my Android device. Image selection on the profile page will need to be tested on iOS. Profile page also needs to be tested for a driver without a profile picture.

Main page shows rider images, with the default image icon for riders without photoLink in backend:

History page shows rider images:

Profile page shows driver image:

Confirmed that uploading an image works for various images.

### Notes 

- Added a temporary person icon as a default profile picture when a user has none; will update if designs change
- When uploading images, they are currently resized to 200 x 200 because otherwise we get an error that the payload is too large. PNG format resizing is not supported by the package we're using right now, but uploading PNGs has seemed to work for various sizes.

### CODE REVIEW

File: lib/models/Driver.dart [hunk 1]
```diff
@@ -6,40 +6,43 @@ class Driver {
...
+  factory Driver.fromJson(Map<String, dynamic> json) {
     return Driver(
         firstName: json['firstName'],
         lastName: json['lastName'],
-        availability: json['availability'],
-        vehicle: json['vehicle']['name'],
         phoneNumber: json['phoneNumber'],
         email: json['email'],
-        photoUrl: photoUrl);
+        photoLink: json['photoLink'] == null ? null : 'http://' + json['photoLink'] + '?dummy=${DateTime.now().millisecondsSinceEpoch}'
```
Comment (tyang98) [inline lib/models/Driver.dart]: What is the purpose of adding the DateTime to the image url?

File: lib/pages/Profile.dart [hunk 2]
```diff
@@ -82,8 +101,12 @@ class _ProfileState extends State<Profile> {
...
                                     child:
                                     Icon(Icons.add, size: _picBtnDiameter),
-                                    onPressed: () {
-                                      // TODO: add functionality to select photo if we decide to store profile images
+                                    onPressed: () async {
+                                      ImagePicker picker = ImagePicker();
+                                      PickedFile pickedFile = await picker.getImage(source: ImageSource.gallery, maxHeight: 200, maxWidth: 200);
+                                      Uint8List bytes = await File(pickedFile.path).readAsBytes();
+                                      String base64Image = base64Encode(bytes);
+                                      driverProvider.updateDriverPhoto(AppConfig.of(context), Provider.of<AuthProvider>(context, listen: false), base64Image);
```
Comment (tyang98) [inline lib/pages/Profile.dart:109]: On my end, when I try to upload a photo locally, I am getting this error: Bad state: Insecure HTTP is not allowed by platform: [http://dti-carriage-staging-public.s3.us-east-2.amazonaws.com/Drivers/37a2ef80-1b9a-11ea-af6b-ab3c592c2075?dummy=1617088252542](http://dti-carriage-staging-public.s3.us-east-2.amazonaws.com/Drivers/37a2ef80-1b9a-11ea-af6b-ab3c592c2075?dummy=1617088252542). The photolink seems to be a correct link.

**Review by tyang98 — COMMENTED** [review]
The code itself looks good to me. I won't approve this yet because there is potentially an issue with uploading and displaying images on my end. It appears each user's photo link has a proper link but Flutter is unable to display it.

File: lib/models/Driver.dart (hunk 1 above)
Comment (helenxuyang) [reply lib/models/Driver.dart]: The NetworkImage that displays the picture won't update if the URL hasn't changed, so adding a unique but useless query at the end just forces it to update (https://stackoverflow.com/questions/49892074/flutter-image-network-not-updated)

File: lib/models/Driver.dart [hunk 3]
```diff
@@ -6,40 +6,43 @@ class Driver {
...
+  factory Driver.fromJson(Map<String, dynamic> json) {
     return Driver(
         firstName: json['firstName'],
         lastName: json['lastName'],
-        availability: json['availability'],
-        vehicle: json['vehicle']['name'],
         phoneNumber: json['phoneNumber'],
         email: json['email'],
-        photoUrl: photoUrl);
+        photoLink: json['photoLink'] == null ? null : 'https://' + json['photoLink'] + '?dummy=${DateTime.now().millisecondsSinceEpoch}'
```
Comment (tyang98) [inline lib/models/Driver.dart:35]: this new 'https://' appears to solve all the previous issues that I was getting

**Review by tyang98 — APPROVED** [review]
I uploaded some images of my driver; created rides with different drivers so I could see their profile picture. Everything seems to be working properly now.

**Pull request merged**